storage/src/key_value/agent/block.rs

```diff
@@ -223,34 +227,49 @@ impl<S: KeyValueStorage + 'static> Agent<S> {
         })
     }
 
-    pub(super) fn get_block_locator_hashes(&mut self) -> Result<Vec<Digest>> {
+    pub(super) fn get_block_locator_hashes(
+        &mut self,
+        points_of_interest: Vec<Digest>,
+        oldest_fork_threshold: usize,
+    ) -> Result<Vec<Digest>> {
         let canon = self.canon()?;
+        let target_height = canon.block_height as u32;
 
         // The number of locator hashes left to obtain; accounts for the genesis block.
-        let mut num_locator_hashes = std::cmp::min(crate::NUM_LOCATOR_HASHES - 1, canon.block_height as u32);
+        let mut num_locator_hashes = std::cmp::min(crate::NUM_LOCATOR_HASHES - 1, target_height);
 
         // The output list of block locator hashes.
-        let mut block_locator_hashes = Vec::with_capacity(num_locator_hashes as usize);
+        let mut block_locator_hashes = Vec::with_capacity(num_locator_hashes as usize + points_of_interest.len());
+
+        for hash in points_of_interest {
+            trace!("block locator hash -- interesting: block# none: {}", hash);
+            block_locator_hashes.push(hash);
+        }
 
         // The index of the current block for which a locator hash is obtained.
-        let mut hash_index = canon.block_height as u32;
+        let mut hash_index = target_height;
 
         // The number of top blocks to provide locator hashes for.
         let num_top_blocks = std::cmp::min(10, num_locator_hashes);
 
         for _ in 0..num_top_blocks {
-            block_locator_hashes.push(self.get_block_hash_guarded(hash_index)?);
+            let hash = self.get_block_hash_guarded(hash_index)?;
+            trace!("block locator hash -- top: block# {}: {}", hash_index, hash);
```

Comment:
do we still need the `trace` logs here? they could get very verbose, so perhaps it would be better to comment them out?

Reply:
it's trace level, and I found them quite useful for diagnosing issues in the field -- considering we don't fetch block locator hashes more than once a sync cycle, I think it's fine